projects/incubator/src/storefrontlib/shared/components/image-zoom/product-view/image-zoom-product-view.component.ts

```diff
@@ -0,0 +1,314 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import { ImageGroup, Product } from '@spartacus/core';
+import {
+  BREAKPOINT,
+  BreakpointService,
+  CurrentProductService,
+  ICON_TYPE,
+} from '@spartacus/storefront';
+import {
+  BehaviorSubject,
+  combineLatest,
+  fromEvent,
+  merge,
+  Observable,
+  of,
+  Subscription,
+} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  shareReplay,
+  switchMap,
+  switchMapTo,
+  tap,
+} from 'rxjs/operators';
+import { ThumbGroup } from '../product-thumbnails/thumbnails.model';
+import {
+  calculatePointerMovePosition,
+  handleOutOfBounds,
+} from './image-zoom-util';
+
+@Component({
+  selector: 'cx-image-zoom-product-view',
+  templateUrl: './image-zoom-product-view.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ImageZoomProductViewComponent implements OnInit, OnDestroy {
+  iconType = ICON_TYPE;
+
+  @Input() galleryIndex: number;
+
+  private mainMediaContainer = new BehaviorSubject<ImageGroup>(null);
+  private defaultImageReady = new BehaviorSubject<boolean>(false);
+  private zoomReady = new BehaviorSubject<boolean>(false);
+  private subscription = new Subscription();
```

Comment:
In `projects/incubator/src/storefrontlib/shared/components/image-zoom/trigger/image-zoom-trigger.component.ts` we marked the subscription as `protected`.
I guess it's better to exposed it to customers. What do you think?

Reply:
Yes true, I missed it